rollup-node.md

```diff
@@ -0,0 +1,285 @@
+
+### From L2 derivation inputs to payload attributes
+
+[payload-attr]: #From-L2-derivation-inputs-to-payload-attributes
+[`PayloadAttributesOPV1`]: #From-L2-derivation-inputs-to-payload-attributes
+
+The rollup reads the following [L2 derivation inputs] for each L1 block:
+
+- L1 block attributes
+   - block number
+   - timestamp
+   - basefee
+- [deposits]
+
+A deposit is an L2 transaction that has been submitted on L1, via a transaction
+sent to the [L2 deposit feed contract][deposit-feed].
+
+While deposits are notably (but not only) used to "deposit" (bridge) ETH and
+tokens to L2, the word *deposit* should be understood as "a transaction
+*deposited* to L2".
+
+The L1 attributes are read from the L1 block header, while deposits are read
+from the block's log entries. Refer to the [**L2 deposit feed contract
+specification**][deposit-feed] for details on how deposits are encoded as
+log entries.
+
+[deposit-feed-spec]: TODO
+
+> **TODO LINK** deposit feed contract specification
+
+From these L2 derivation inputs, the rollup node constructs an expanded version
+of the [Engine API PayloadAttributesV1 object][PayloadAttributesV1]:
+
+[PayloadAttributesV1]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1
+
+```
+PayloadAttributesOPV1: {
+	timestamp: QUANTITY
+	random: DATA (32 bytes)
+	suggestedFeeRecipient: DATA (20 bytes)
+	transactions: array of DATA
+}
+```
+
+The type notation used here refers to the [HEX value encoding] used by the
+[Ethereum JSON-RPC API specification][JSON-RPC-API], as this structure will need to
+be sent over JSON-RPC. `array` refers to a JSON array.
+
+[HEX value encoding]: https://eth.wiki/json-rpc/API#hex-value-encoding
+[JSON-RPC-API]: https://eth.wiki/json-rpc/API
+
+The object properties must be set as follows:
+
+- `timestamp` is set to the current [unix time] (number of elapsed seconds since
+  00:00:00 UTC on 1 January 1970), rounded to the closest multiple of 2 seconds.
+- `random` is currently set to all zero bytes. After [the merge][merge], the randomness
+  value exposed for the L1 block will be copied in this property.
+- `suggestedFeeRecipient` is set to an address where the sequencer would like to
+  direct the fees
+- `transactions` is an array of transactions, RLP-encoded in the [EIP-2718]
+  format (i.e. as a sequence of a single byte and a byte array), of the allowed
+  types. We specify how to select the transactions below.
+
+[unix type]: https://en.wikipedia.org/wiki/Unix_time
+[merge]: https://ethereum.org/en/eth2/merge/
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+> For details on RLP encoding, refer to the yellow paper, or look at [this
+> hyperlinked implementation][encode-tx].
+
+[encode-tx]: https://github.com/norswap/nanoeth/blob/cc5d94a349c90627024f3cd629a2d830008fec72/src/com/norswap/nanoeth/transactions/Transaction.java#L84-L130
+
+In particular, we allow the following two transaction types:
```

Comment:
Is it intentional to block type 1 transactions? The ones with the accesslist that aren't eip1559

Reply:
I figured it makes for easier spec and easier implementation, and shouldn't be an impediment at the wallet level (since they needed to update for us anyway, after EIP-1559).

That being said, maybe there are some legacy things we do absolutely wish to support or that will byte us in the ass if we don't?

I'm thinking that over the long run Ethereum is going to deprecate the non EIP-1559 transactions anyway (similarly to what is already underway with pre-EIP-155 transactions that are being soft-deprecated by Geth (there might be an EIP?)).

Reply:
On further reflection, I think it's a mistake to try to constrain the type of transactions that are allowed here. See proto's comments to that effect: https://github.com/ethereum-optimism/optimistic-specs/pull/43#discussion_r759678629